File: test-suite/src/test/java/io/micronaut/servlet/docs/websocket/jakarta/EchoClient.java
package io.micronaut.servlet.docs.websocket.jakarta;

import io.micronaut.context.annotation.Requires;
import io.micronaut.websocket.annotation.ClientWebSocket;
import io.micronaut.websocket.annotation.OnMessage;

import java.util.Collection;
import java.util.concurrent.ConcurrentLinkedQueue;

@Requires(property = "spec.name", value = "EchoEndpointSpec")
@ClientWebSocket("/ws/echo/{room}")
public abstract class EchoClient implements AutoCloseable {

    private final Collection<String> replies = new ConcurrentLinkedQueue<>();

    @OnMessage
    public void onMessage(String message) {
        replies.add(message);
    }

    public abstract void send(String message);

    public Collection<String> getReplies() {
        return replies;
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/servlet/docs/websocket/jakarta/EchoClient.kt
package io.micronaut.servlet.docs.websocket.jakarta

import io.micronaut.context.annotation.Requires
import io.micronaut.websocket.annotation.ClientWebSocket
import io.micronaut.websocket.annotation.OnMessage
import java.util.concurrent.ConcurrentLinkedQueue

@Requires(property = "spec.name", value = "EchoEndpointSpec")
@ClientWebSocket("/ws/echo/{room}")
abstract class EchoClient : AutoCloseable {

    val replies: MutableCollection<String> = ConcurrentLinkedQueue()

    @OnMessage
    fun onMessage(message: String) {
        replies.add(message)
    }

    abstract fun send(message: String)
}
